src/components/ui/Button.tsx

```diff
@@ -0,0 +1,51 @@
+import { cva, type VariantProps } from 'class-variance-authority';
+
+// cva for button styles
+const buttonStyles = cva(
+  'py-3 px-8 rounded-full font-bold text-lg transition-all duration-300 cursor-pointer', 
+  {
+    variants: {
+      variant: {
+        filled: '',
+        outline: 'bg-transparent',
+      },
+      colorScheme: {
+        blue: 'text-white bg-blue-600 border-none hover:bg-blue-700 hover:text-white',
+        white: 'text-blue-600 bg-white border-none hover:bg-gray-200',
+      },
+    },
+    compoundVariants: [
+      {
+        variant: 'outline',
+        colorScheme: 'blue',
+        className: 'border-2 border-blue-600 text-blue-600 hover:bg-blue-600 hover:text-white',
+      },
+      {
+        variant: 'outline',
+        colorScheme: 'white',
+        className: 'border-2 border-white text-white hover:bg-white hover:text-blue-600',
+      },
+    ],
+    defaultVariants: {
+      variant: 'filled',
+      colorScheme: 'blue',
+    },
+  }
+);
+
+// button props
+interface ButtonProps extends VariantProps<typeof buttonStyles> {
+  text: string;
+  onClick?: () => void;
+}
+
+// button component using CVA
+const Button: React.FC<ButtonProps> = ({ text, variant, colorScheme, onClick }) => {
+  return (
+    <button onClick={onClick} className={buttonStyles({ variant, colorScheme })}>
+      {text}
+    </button>
+  );
+};
+
+export default Button;
```

Comment:
One small addition lang for reusability of this component. We might want to cover the use case na mag nest tayo ng link inside a button. Can you add an asChild prop na murag ani pag structure:


```
import { Slot } from '@radix-ui/react-slot';

function Button({ asChild, ...props }) {
  const Comp = asChild ? Slot : 'button';
  return <Comp {...props} />;
}

```

Refer to this link for more info: https://www.radix-ui.com/primitives/docs/utilities/slot
cc: @shunindesuuu

Reply:
Include a ref prop nalang pud para incase need e access sa DOM ang element